File: software/examples/python/example_callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_accelerometer import BrickletAccelerometer

# Callback function for acceleration callback (parameters have unit g/1000)
def cb_acceleration(x, y, z):
    print('Acceleration(X): ' + str(x/1000.0) + ' g')
    print('Acceleration(Y): ' + str(y/1000.0) + ' g')
    print('Acceleration(Z): ' + str(z/1000.0) + ' g')
    print('')

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    a = BrickletAccelerometer(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected
    
    # Set Period for acceleration callback to 0.1s (100ms)
    # Note: The callback is only called every second if the 
    #       acceleration has changed since the last call!
    a.set_acceleration_callback_period(100)

    # Register acceleration callback to function cb_acceleration
    a.register_callback(a.CALLBACK_ACCELERATION, cb_acceleration)
    
    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/perl/example_callback.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletAccelerometer;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change to your UID

# Callback subroutine for acceleration callback (parameters have unit g/1000)
sub cb_acceleration
{
    my ($x, $y, $z) = @_;

    print "Acceleration[X]: " . $x/1000.0 . " g\n";
    print "Acceleration[Y]: " . $y/1000.0 . " g\n";
    print "Acceleration[Z]: " . $z/1000.0 . " g\n";
    print "\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $a = Tinkerforge::BrickletAccelerometer->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register acceleration callback to subroutine cb_acceleration
$a->register_callback($a->CALLBACK_ACCELERATION, 'cb_acceleration');

# Set period for acceleration callback to 1s (1000ms)
# Note: The acceleration callback is only called every second
#       if the acceleration has changed since the last call!
$a->set_acceleration_callback_period(1000);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
